controllers/api_keys.rb

```diff
@@ -0,0 +1,46 @@
+class CalendarEvents::ApiKeysController < ApplicationController
```

Comment:
@haveyaseen @fzngagan Why aren't we using or modifying the existing user_api_keys_controller here? Whenever possible it's better to use, or build on top of, existing Discourse structures rather than build our own.

Reply:
I figured that wasn't the right place as the `user_api_keys_controller` is actually meant for third party software to generate user API keys for their own access on behalf of the user. It uses RSA to encrypt the returned API key.

That's a much different use case than we're having here. In plugins we actually have access to Discourse's models and database so we can avoid needlessly fiddling with public keys and simply generate the required `UserApiKey` record ourselves. In my opinion this is much cleaner, especially because Discourse does not use its `user_api_keys_controller` internally either. So monkey-patching it wouldn't even allow us to write less JavaScript code.

In a way one could argue that we are abusing that feature here, but it still makes sense to use it as it allows us to easily restrict the scope of the keys. Should they be leaked, an attacker will not gain access to other functions apart from the feeds of categories that the user has access to.